src/lib/functions/registry.js

```diff
@@ -128,8 +117,27 @@ class FunctionsRegistry {
         return runtime.onDirectoryScan({ directory })
       }),
     )
+  }
+
+  registerFunction(name, func) {
+    if (func.isBackground && !this.capabilities.backgroundFunctions) {
+      this.logger.warn(getLogMessage('functions.backgroundNotSupported'))
+    }
+
+    this.functions.set(name, func)
+    this.buildFunctionAndWatchFiles(func)
+
+    this.logger.log(`${NETLIFYDEVLOG} ${chalk.green('Loaded')} function ${chalk.yellow(name)}.`)
+  }
+
+  async scan(directories) {
+    if (directories.length === 0) {
```

Comment:
I think we still need to filter `null` directories as `getInternalFunctionsDir` can return `null` and it is used in https://github.com/netlify/cli/pull/2954/files#diff-de106b743be68ae2e9146d09931f894e258783a68db916a6d0e39e3908b28c84R165

Reply:
Good catch. Done in https://github.com/netlify/cli/pull/2954/commits/d1eb8ca56a6ba0890feda9e5773e3e41fa684538.